### PR TITLE
Replaces sleeping related Mixins with somnus

### LINKED ISSUE #21: (Crash) Somnus not compatible
```
Failed to start Minecraft:
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'paradoxconfig'!
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:53)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:36)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:137)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)
	... 8 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at net.minecraft.class_2246.<clinit>(class_2246.java:99)
	at net.minecraft.class_3523.<clinit>(class_3523.java:13)
	at net.minecraft.class_5471.<clinit>(class_5471.java:11)
	at net.minecraft.class_5458.method_30573(class_5458.java:44)
	at net.minecraft.class_5458.method_30566(class_5458.java:75)
	at java.util.LinkedHashMap.forEach(Unknown Source)
	at net.minecraft.class_5458.<clinit>(class_5458.java:74)
	at net.minecraft.class_2378.<clinit>(class_2378.java:237)
	at io.github.cottonmc.jankson.JanksonFactory.builder(JanksonFactory.java:85)
	at io.github.redstoneparadox.paradoxconfig.codec.JanksonConfigCodec.<init>(JanksonConfigCodec.java:18)
	at io.github.redstoneparadox.paradoxconfig.ParadoxConfig.onPreLaunch(ParadoxConfig.java:21)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50)
	... 11 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector allowVampiresToSleepInDay(Lnet/minecraft/class_1937;)Z in haema.mixins.json:PlayerEntityMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap haema-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 29 more
Exiting with -1
```

### PR DESCRIPTION
**Pull request by tyra314**
This replaces the sleeping related Mixins, which were not compatible with Somnus, with Somnus itself. Redirect Mixins are inherently not compatible with each other, so all Mixins would have to be rewritten as @inject or something. Seems easier this way. If you can't beat them, join them, right?

Fixes #21

### CODE REVIEW

**Comment by williambl** [comment]
Ah awesome, thanks!

File: src/main/kotlin/com/williambl/haema/Haema.kt [hunk 1]
```diff
@@ -142,7 +145,33 @@ fun init() {
...
+
+    PlayerSleepEvents.TRY_SLEEP.register(PlayerSleepEvents.TrySleep { player, pos ->
+        if (player is Vampirable) {
+            if (player.isVampire && !player.world.isDay) {
+                return@TrySleep PlayerEntity.SleepFailureReason.NOT_POSSIBLE_NOW
+            }
+        }
+
+        null
+    })
```
Comment (williambl) [inline src/main/kotlin/com/williambl/haema/Haema.kt:167]: Vampires can sleep in day too :)
```suggestion
```

**Review by williambl — CHANGES_REQUESTED** [review]
Thanks so much, just one little thing :)

File: src/main/kotlin/com/williambl/haema/Haema.kt (hunk 1 above)
Comment (tyra314) [reply src/main/kotlin/com/williambl/haema/Haema.kt]: I wasn't aware of that. Fixed it.

**Review by williambl — APPROVED** [review]

**Pull request merged**